icefoganalytics/elcc-data-management
# ELCC26: Navigation Enhancement

Fixes https://yg-hpw.atlassian.net/browse/ELCC-26

# Context

Is there an easier way to move down to the next cell to fill? Tab you have to go through est first, enter and arrows don’t work.

It sounds like there are limited ways of moving in and out of cells (TAB).  Is it possible to enable the ability to use enter and arrows to navigate?  Currently it is taking a considerable amount of time to navigate screens to enter data.

# Implementation

1. Add listeners for Enter, Shift+Enter, ArrowUp, ArrowDown, ArrowRight, and ArrowLeft events to Worksheet inputs. See screenshot for key functions enabled.
2. Split some worksheet components into smaller components.
3. Standardize some component locations.
4. Fix linting in /web directory (front-end)
5. Pull `dev edit-pr <github-pr-url>` helper from Travel Auth project to get AI auto-completion as I type PR instructions.

# Screenshots

Keyboard Shortcuts Modal
![image](https://github.com/user-attachments/assets/11829b3e-c5f4-49ab-9077-089247c9fb01)

# Testing Instructions

1. Boot the app via `dev up`
2. Check that the app complies, and that you can log in at http://localhost:8080.
3. Click on the Education is Currently Funding card.
4. Double click on a Child Care Centre to go to the edit page.
6. Click on the Worksheets tab.
7. Add worksheets if there aren't any available.
8. Click on the keyboard icon at the top right of any section header to see the list of shortcuts.
9. Click on the first row of the table.
10. Press Enter to move down the column, selecting the text each time.
11. Press Enter until you jump to the next section.
12. Press Shift+Enter to move up the column, selecting the text each time.
13. Note that wehn you reach the top of the column, it stops changing selected input.
14. Try the same thing with ArrowUp, ArrowDown.
15. Enter some content in the first cell.
16. Press ArrowRight to move to the next column.
17. Note that the text is selected.
18. Press ArrowLeft to move to the start of the input.
19. Press ArrowLeft again to move to the previous column. Note that the text is selected.
20. Press ArrowRight to change input to end of selected text.
21. Note that Tab and Shift+Tab still work as expected.
